mesonbuild/backend/xcodebackend.py

```diff
@@ -53,6 +53,35 @@
                 }
 BOOL2XCODEBOOL = {True: 'YES', False: 'NO'}
 LINKABLE_EXTENSIONS = {'.o', '.a', '.obj', '.so', '.dylib'}
+XCODEVERSIONS = {'1400': ('Xcode 14.0', 56),
+                 '1300': ('Xcode 13.0', 55),
+                 '1200': ('Xcode 12.0', 54),
+                 '1140': ('Xcode 11.4', 53),
+                 '1100': ('Xcode 11.0', 52),
+                 '1000': ('Xcode 10.0', 51),
+                 '930': ('Xcode 9.3', 50),
+                 '800': ('Xcode 8.0', 48),
+                 '630': ('Xcode 6.3', 47),
+                 '320': ('Xcode 3.2', 46),
+                 '310': ('Xcode 3.1', 45)
+                 }
+
+def autodetect_xcode_version() -> T.Tuple[str, int]:
+    try:
+        pc = subprocess.Popen(['xcodebuild', '-version'], encoding='utf-8', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except FileNotFoundError:
+        raise MesonException('Could not detect Xcode. Please install it if you wish to use the Xcode backend.')
+    stdout, stderr = pc.communicate()
+    if pc.returncode != 0:
+        raise MesonException(f'An error occurred while detecting Xcode: {stderr}')
```

Comment:
In theory if it is ever possible for xcodebuild to return non-utf8 data here, mesonlib.Popen_safe is useful.